File: 프로그래머스/lv0/120848. 팩토리얼/팩토리얼.py
def solution(n):
    
    def f(x):
        if x <= 1:
            return 1
        else:
            return x*f(x-1)
    
    for i in range(1, 11):
        if f(i) == n:
            return i
        elif f(i)> n:
            return i-1
    

File: 프로그래머스/0/120848. 팩토리얼/팩토리얼.java
class Solution {
    public int solution(int x) {
        int n = 1; 
        int factorial = 1; 
        
        while (factorial < x) {
            n++; 
            factorial *= n; 
        }
        
        if (factorial == x) {
            return n;
        } 
        return n-1;
    }
}
